### PR TITLE
Remove BOM from json files

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #276 

1. Removed BOM from json files. 
2. Added a test to check if BOM present

The change does not show the BOM removed but I ran the test before and after and BOM has been removed from the below files.

### CODE REVIEW

File: test/Templates.Test/ByteOrderMarkTest.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        private readonly ITestOutputHelper _output;
+
+        public ByteOrderMarkTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Theory]
+        [InlineData("\\Microsoft.AspNetCore.SpaTemplates\\content")]
```
Comment (Eilon) [inline test/Templates.Test/ByteOrderMarkTest.cs]: Might want to use `@"..."` strings here too.

File: test/Templates.Test/ByteOrderMarkTest.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        [InlineData("\\Microsoft.DotNet.Web.ProjectTemplates\\content")]
+        [InlineData("\\Microsoft.DotNet.Web.Spa.ProjectTemplates\\content")]
+        public void CheckForByteOrderMarkSpaTemplates(string path)
+        {
+            var currentDirectory = Directory.GetCurrentDirectory();
+            var srcDirectory = Path.GetFullPath(Path.Combine(currentDirectory, @"..\..\..\..\..\src"));
+            var directories = Directory.GetDirectories(srcDirectory + path, "*Sharp");
+            var checkFiles = false;
```
Comment (Eilon) [inline test/Templates.Test/ByteOrderMarkTest.cs]: Not sure I understand the semantics of this variable. Can we find a better name for it?

File: test/Templates.Test/ByteOrderMarkTest.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            foreach (var directory in directories)
+            {
+                var files = Directory.GetFiles(directory, "*.json");
+                foreach (var file in files)
+                {
+                    var filePath = Path.GetFullPath(file);
+                    var fileStream = new FileStream(filePath, FileMode.Open);
+
+                    var bits = new byte[3];
```
Comment (Eilon) [inline test/Templates.Test/ByteOrderMarkTest.cs]: Bytes, not bits.

File: test/Templates.Test/ByteOrderMarkTest.cs (hunk 2 above)
Comment (jbagga) [reply test/Templates.Test/ByteOrderMarkTest.cs]: Basically a flag that says the test output should be checked for files that have BOM characters.

Comment (Eilon) [reply test/Templates.Test/ByteOrderMarkTest.cs]: Isn't it saying the *result* of the test (success/failure), as opposed to whether something should be checked?

File: test/Templates.Test/ByteOrderMarkTest.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        [InlineData(@"\Microsoft.AspNetCore.SpaTemplates\content")]
+        [InlineData(@"\Microsoft.DotNet.Web.ProjectTemplates\content")]
+        [InlineData(@"\Microsoft.DotNet.Web.Spa.ProjectTemplates\content")]
+        public void CheckForByteOrderMarkSpaTemplates(string path)
+        {
+            var currentDirectory = Directory.GetCurrentDirectory();
+            var srcDirectory = Path.GetFullPath(Path.Combine(currentDirectory, @"..\..\..\..\..\src"));
+            var directories = Directory.GetDirectories(srcDirectory + path, "*Sharp");
+
+            var filesWithBOMCharactersPresent = false;
```
Comment (jbagga) [inline test/Templates.Test/ByteOrderMarkTest.cs]: Is this any better? @Eilon

Comment (Eilon) [inline test/Templates.Test/ByteOrderMarkTest.cs]: Ah, new name is better!

**Comment by jbagga** [comment]
Yes I have confirmed that all the files in this PR were reported and the test failed. It now passes with the BOM removed

**Pull request merged**